File: orderApi/models/orden.cs
//modelo de orden




namespace orderApi.models
{
    public class Orden
    {
        public int Id { get; set; }
        public int PacienteId { get; set; }
        public DateTime FechaAtencion { get; set; }

        // Relación con paciente
        public Paciente Paciente { get; set; } = null!;

        // Relación con examenes
        public List<Examen> Examenes { get; set; } = new List<Examen>();
    }
}

File: ordenes_app/lib/models/orden.dart


import 'package:json_annotation/json_annotation.dart';
import 'package:ordenes_app/models/examen.dart';

part 'orden.g.dart';

@JsonSerializable()
class Orden{
  final int id;
  final DateTime fechaAtencion;
  final List<Examen> examenes;

  Orden({
    required this.id,
    required this.fechaAtencion,
    required this.examenes,
  });

  factory Orden.fromJson(Map<String, dynamic> json) => _$OrdenFromJson(json);

  Map<String, dynamic> toJson() => _$OrdenToJson(this);
}